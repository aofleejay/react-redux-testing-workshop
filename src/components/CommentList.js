import React from 'react'
import CommentItem from './CommentItem'

const CommentList = ({ comments, loadMore }) => (
  <div>
    <div>
      {
        comments.map(({ id, title, body, author, postat }) => (
          <CommentItem
            key={id}
            title={title}
            body={body}
            author={author}
            postat={postat}
          />
        ))
      }
    </div>
    <button
      id="load-more"
      className="button is-primary load-more"
      onClick={loadMore}
    >
      Load More
    </button>
  </div>
)

export default CommentList
