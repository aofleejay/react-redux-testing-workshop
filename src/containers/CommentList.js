import React, { Component } from 'react'
import { connect } from 'react-redux'
import CommentList from '../components/CommentList'
import { getComments } from '../actions/comments'

export class CommentListContainer extends Component {
  componentDidMount() {
    this.props.getComments()
  }

  render() {
    return (
      <CommentList 
        comments={this.props.comments}
        loadMore={this.props.getComments}
      />
    )
  }
}

export const mapStateToProps = state => ({
  comments: state.comments,
})

export const mapDispatchToProps = dispatch => ({
  getComments: () => dispatch(getComments()),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(CommentListContainer)
