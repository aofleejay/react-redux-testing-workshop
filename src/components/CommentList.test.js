import React from 'react'
import { shallow } from 'enzyme'
import CommentList from './CommentList'

jest.mock('../components/CommentItem', () => 'CommentItem')

describe('Test CommentList component', () => {
  const props = {
    comments: [
      { id: 1, title: 'title1', body: 'body1', author: 'author1', postat: '1m ago' },
      { id: 2, title: 'title2', body: 'body2', author: 'author2', postat: '2m ago' },
    ],
    loadMore: jest.fn(),
  }
  let wrapper
  
  beforeEach(() => {
    wrapper = shallow(<CommentList {...props} />)
  })

  it('Should match its snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('Should call loadMore props when click load more button', () => {
    wrapper.find('#load-more').simulate('click')
    expect(props.loadMore).toHaveBeenCalledTimes(1)
  })
})
