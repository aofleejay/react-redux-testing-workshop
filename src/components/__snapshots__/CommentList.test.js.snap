// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CommentList component Should match its snapshot 1`] = `
<div>
  <div>
    <CommentItem
      author="author1"
      body="body1"
      key="1"
      postat="1m ago"
      title="title1"
    />
    <CommentItem
      author="author2"
      body="body2"
      key="2"
      postat="2m ago"
      title="title2"
    />
  </div>
  <button
    className="button is-primary load-more"
    id="load-more"
    onClick={[MockFunction]}
  >
    Load More
  </button>
</div>
`;
